"View dependency licenses" makes Zed crash

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

On macOSm select Help -> View dependency licenses, and kaboom.

### Environment

Zed: v0.166.0 (Zed Dev 0511768b226094a25fddb99fc6e7ef53ad220197)
OS: macOS 15.2.0
Memory: 24 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2024-12-05T23:45:12.688797+01:00 [INFO] ========== starting zed ==========
2024-12-05T23:45:12.712001+01:00 [INFO] Opening main db
2024-12-05T23:45:12.713+01:00 [INFO] Opening main db
2024-12-05T23:45:12.715043+01:00 [INFO] Using git binary path: Some("/Applications/Zed Dev.app/Contents/MacOS/git")
2024-12-05T23:45:12.804711+01:00 [INFO] set environment variables from shell:/bin/zsh, path:/Users/mocenigo/.zi/polaris/sbin:/Users/mocenigo/.zi/polaris/bin:/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/Users/mocenigo/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin://Applications/Topaz Photo AI.app/Contents/Resources/bin:/Library/Apple/usr/bin:/Library/TeX/texbin:/Users/mocenigo/Library/Python/3.8/bin:/opt/homebrew/opt/llvm/bin:/Users/mocenigo/.local/bin:/Users/mocenigo/.ghc/bin:/Users/mocenigo/.cabal/bin:/Users/mocenigo/.ghcup/bin:/opt/homebrew/opt/icu4c/sbin:/opt/homebrew/opt/icu4c/bin:/Users/mocenigo/.cache/lm-studio/bin
2024-12-05T23:45:12.888554+01:00 [ERROR] theme not found: Melange Modified Light
2024-12-05T23:45:12.889154+01:00 [INFO] No prompt template overrides directory found at /Users/mocenigo/.config/zed/prompt_overrides. Using built-in prompts.
2024-12-05T23:45:12.889415+01:00 [INFO] extensions updated. loading 15, reloading 0, unloading 0
2024-12-05T23:45:12.929992+01:00 [INFO] Opening main db
2024-12-05T23:45:12.936159+01:00 [ERROR] theme not found: Melange Modified Light
2024-12-05T23:45:12.937871+01:00 [ERROR] theme not found: Melange Modified Light
2024-12-05T23:45:12.940651+01:00 [WARN] Theme "macOS Classic Light" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-05T23:45:12.94072+01:00 [WARN] Theme "macOS Classic Dark" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-05T23:45:12.940778+01:00 [WARN] Theme "macOS Classic Dark2" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-05T23:45:12.970088+01:00 [INFO] set status on client 0: Authenticating
2024-12-05T23:45:12.971574+01:00 [INFO] Opening main db
2024-12-05T23:45:12.974505+01:00 [INFO] Opening main db
2024-12-05T23:45:12.976222+01:00 [INFO] set status on client 137487: Connecting
2024-12-05T23:45:13.087748+01:00 [INFO] using project environment variables shell launched in "/Users/mocenigo/Research/PAC-Analysis/paper". PATH="/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/Users/mocenigo/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin://Applications/Topaz Photo AI.app/Contents/Resources/bin:/Library/Apple/usr/bin:/Library/TeX/texbin:/Users/mocenigo/Library/Python/3.8/bin:/opt/homebrew/opt/llvm/bin:/Users/mocenigo/.local/bin:/Users/mocenigo/.ghc/bin:/opt/homebrew/sbin:/opt/homebrew/opt/icu4c/sbin:/opt/homebrew/opt/icu4c/bin:/opt/homebrew/bin:/Users/mocenigo/bin:/Users/mocenigo/.zi/polaris/sbin:/Users/mocenigo/.zi/polaris/bin:/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/Users/mocenigo/.cabal/bin:/Users/mocenigo/.ghcup/bin:/Users/mocenigo/.cache/lm-studio/bin:/Users/mocenigo/.cache/lm-studio/bin"
2024-12-05T23:45:13.70942+01:00 [WARN] no language server found matching 'texlab'
2024-12-05T23:45:13.710427+01:00 [WARN] no language server found matching 'texlab'
2024-12-05T23:45:13.711546+01:00 [WARN] no language server found matching 'texlab'
2024-12-05T23:45:13.711709+01:00 [WARN] no language server found matching 'texlab'
2024-12-05T23:45:13.711745+01:00 [WARN] no language server found matching 'texlab'
2024-12-05T23:45:13.711783+01:00 [WARN] no language server found matching 'texlab'
2024-12-05T23:45:13.724199+01:00 [INFO] starting language server process. binary path: "/opt/homebrew/bin/node", working directory: "/", args: ["/Users/mocenigo/Library/Application Support/Zed/copilot/copilot-v0.7.0/dist/language-server.js", "--stdio"]
2024-12-05T23:45:13.941598+01:00 [INFO] attempting to start language server "texlab", path: "/Users/mocenigo/Research/PAC-Analysis/paper", id: 1
2024-12-05T23:45:14.007576+01:00 [INFO] starting language server process. binary path: "/opt/homebrew/bin/texlab", working directory: "/Users/mocenigo/Research/PAC-Analysis/paper", args: []
2024-12-05T23:45:14.14038+01:00 [WARN] unhandled capability registration: Registration { id: "pull-config", method: "workspace/didChangeConfiguration", register_options: None }
2024-12-05T23:45:14.190768+01:00 [INFO] Language server with id 0 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "[lsp] GitHub Copilot Language Server 1.236.0 initialized"
}
2024-12-05T23:45:14.209694+01:00 [INFO] Language server with id 0 sent unhandled notification workspace/configuration:
{
  "items": [
    {
      "section": "github.copilot"
    },
    {
      "section": "github-enterprise"
    },
    {
      "section": "http"
    },
    {
      "section": "telemetry"
    }
  ]
}
2024-12-05T23:45:25.694508+01:00 [INFO] ========== starting zed ==========
2024-12-05T23:45:25.723801+01:00 [INFO] Opening main db
2024-12-05T23:45:25.725021+01:00 [INFO] Opening main db
2024-12-05T23:45:25.727767+01:00 [INFO] Using git binary path: Some("/Applications/Zed Dev.app/Contents/MacOS/git")
2024-12-05T23:45:25.838597+01:00 [INFO] set environment variables from shell:/bin/zsh, path:/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/Users/mocenigo/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin://Applications/Topaz Photo AI.app/Contents/Resources/bin:/Library/Apple/usr/bin:/Library/TeX/texbin:/Users/mocenigo/Library/Python/3.8/bin:/opt/homebrew/opt/llvm/bin:/Users/mocenigo/.local/bin:/Users/mocenigo/.ghc/bin:/opt/homebrew/sbin:/opt/homebrew/opt/icu4c/sbin:/opt/homebrew/opt/icu4c/bin:/opt/homebrew/bin:/Users/mocenigo/bin:/Users/mocenigo/Research-PyEnv/bin:/Users/mocenigo/.zi/polaris/sbin:/Users/mocenigo/.zi/polaris/bin:/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/Users/mocenigo/.cabal/bin:/Users/mocenigo/.ghcup/bin:/Users/mocenigo/.cache/lm-studio/bin:/Users/mocenigo/.cache/lm-studio/bin
2024-12-05T23:45:25.903497+01:00 [ERROR] theme not found: Melange Modified Light
2024-12-05T23:45:25.904078+01:00 [INFO] No prompt template overrides directory found at /Users/mocenigo/.config/zed/prompt_overrides. Using built-in prompts.
2024-12-05T23:45:25.904326+01:00 [INFO] extensions updated. loading 15, reloading 0, unloading 0
2024-12-05T23:45:25.940303+01:00 [INFO] Opening main db
2024-12-05T23:45:25.947092+01:00 [ERROR] theme not found: Melange Modified Light
2024-12-05T23:45:25.951437+01:00 [WARN] Theme "macOS Classic Light" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-05T23:45:25.951512+01:00 [WARN] Theme "macOS Classic Dark" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-05T23:45:25.951574+01:00 [WARN] Theme "macOS Classic Dark2" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-05T23:45:25.989677+01:00 [INFO] set status on client 0: Authenticating
2024-12-05T23:45:25.989998+01:00 [INFO] Opening main db
2024-12-05T23:45:25.997267+01:00 [INFO] set status on client 137487: Connecting
2024-12-05T23:45:25.998288+01:00 [INFO] Opening main db
2024-12-05T23:45:26.024792+01:00 [INFO] using project environment variables from CLI. PATH="/Users/mocenigo/Research-PyEnv/bin:/Users/mocenigo/.zi/polaris/sbin:/Users/mocenigo/.zi/polaris/bin:/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/Users/mocenigo/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin://Applications/Topaz Photo AI.app/Contents/Resources/bin:/Library/Apple/usr/bin:/Library/TeX/texbin:/Users/mocenigo/Library/Python/3.8/bin:/opt/homebrew/opt/llvm/bin:/Users/mocenigo/.local/bin:/Users/mocenigo/.ghc/bin:/Users/mocenigo/.cabal/bin:/Users/mocenigo/.ghcup/bin:/opt/homebrew/opt/icu4c/sbin:/opt/homebrew/opt/icu4c/bin:/Users/mocenigo/.cache/lm-studio/bin"
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -322,9 +322,6 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Generate license file
-        run: script/generate-licenses
-
       - name: Create macOS app bundle
         run: script/bundle-mac
 

--- a/.github/workflows/release_nightly.yml
+++ b/.github/workflows/release_nightly.yml
@@ -86,9 +86,6 @@ jobs:
           echo "Publishing version: ${version} on release channel nightly"
           echo "nightly" > crates/zed/RELEASE_CHANNEL
 
-      - name: Generate license file
-        run: script/generate-licenses
-
       - name: Create macOS app bundle
         run: script/bundle-mac
 

--- a/script/bundle-mac
+++ b/script/bundle-mac
@@ -84,6 +84,9 @@ version_info=$(rustc --version --verbose)
 host_line=$(echo "$version_info" | grep host)
 local_target_triple=${host_line#*: }
 
+# Generate the licenses first, so they can be baked into the binaries
+script/generate-licenses
+
 if [ "$local_arch" = true ]; then
     echo "Building for local target only."
     cargo build ${build_flag} --package zed --package cli --package remote_server

